diff --git a/gradle/wrapper/gradle-wrapper.jar b/gradle/wrapper/gradle-wrapper.jar
index 8c0fb64..13372ae 100644
Binary files a/gradle/wrapper/gradle-wrapper.jar and b/gradle/wrapper/gradle-wrapper.jar differ
diff --git a/gradlew b/gradlew
index 91a7e26..9d82f78 100755
--- a/gradlew
+++ b/gradlew
@@ -42,11 +42,6 @@ case "`uname`" in
     ;;
 esac
 
-# For Cygwin, ensure paths are in UNIX format before anything is touched.
-if $cygwin ; then
-    [ -n "$JAVA_HOME" ] && JAVA_HOME=`cygpath --unix "$JAVA_HOME"`
-fi
-
 # Attempt to set APP_HOME
 # Resolve links: $0 may be a link
 PRG="$0"
@@ -61,9 +56,9 @@ while [ -h "$PRG" ] ; do
     fi
 done
 SAVED="`pwd`"
-cd "`dirname \"$PRG\"`/" >&-
+cd "`dirname \"$PRG\"`/" >/dev/null
 APP_HOME="`pwd -P`"
-cd "$SAVED" >&-
+cd "$SAVED" >/dev/null
 
 CLASSPATH=$APP_HOME/gradle/wrapper/gradle-wrapper.jar
 
@@ -114,6 +109,7 @@ fi
 if $cygwin ; then
     APP_HOME=`cygpath --path --mixed "$APP_HOME"`
     CLASSPATH=`cygpath --path --mixed "$CLASSPATH"`
+    JAVACMD=`cygpath --unix "$JAVACMD"`
 
     # We build the pattern for arguments to be converted via cygpath
     ROOTDIRSRAW=`find -L / -maxdepth 1 -mindepth 1 -type d 2>/dev/null`
diff --git a/library/build.gradle b/library/build.gradle
index beb4b64..35db2d9 100644
--- a/library/build.gradle
+++ b/library/build.gradle
@@ -25,88 +25,89 @@ dependencies {
   compile 'com.android.support:support-annotations:21.0.3'
 }
 
-apply plugin: 'com.github.dcendents.android-maven'
-apply plugin: 'com.jfrog.bintray'
-
-def repo = new File(rootDir, "repository")
-
-apply plugin: 'maven'
-
-uploadArchives {
-  repositories {
-    mavenDeployer {
-      repository url: "file://${repo.absolutePath}"
-      pom.version = VERSION_NAME // version
-      pom.groupId = GROUP // グループ名
-      pom.artifactId = POM_ARTIFACT_ID // ライブラリ名
-    }
-  }
-}
-
-version = "1.2.0"
-def siteUrl = 'https://github.com/baoyongzhang/android-PullRefreshLayout'
-def gitUrl = 'https://github.com/baoyongzhang/android-PullRefreshLayout.git'
-group = "com.baoyz.pullrefreshlayout"
-install {
-  repositories.mavenInstaller {
-    // This generates POM.xml with proper parameters
-    pom {
-      project {
-        packaging 'aar'
-        // Add your description here
-        name 'This component like SwipeRefreshLayout'
-        url siteUrl
-        // Set your license
-        licenses {
-          license {
-            name 'The MIT License (MIT)'
-            url 'http://baoyz.com/licenses/LICENSE.txt'
-          }
-        }
-        developers {
-          developer {
-            id 'baoyongzhang'
-            name 'baoyongzhang'
-            email 'baoyz94@gmail.com'
-          }
-        }
-        scm {
-          connection gitUrl
-          developerConnection gitUrl
-          url siteUrl
-        }
-      }
-    }
-  }
-}
-task sourcesJar(type: Jar) {
-  from android.sourceSets.main.java.srcDirs
-  classifier = 'sources'
-}
-task javadoc(type: Javadoc) {
-  source = android.sourceSets.main.java.srcDirs
-  classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
-}
-task javadocJar(type: Jar, dependsOn: javadoc) {
-  classifier = 'javadoc'
-  from javadoc.destinationDir
-}
-artifacts {
-  archives javadocJar
-  archives sourcesJar
-}
-Properties properties = new Properties()
-properties.load(project.rootProject.file('local.properties').newDataInputStream())
-bintray {
-  user = properties.getProperty("bintray.user")
-  key = properties.getProperty("bintray.apikey")
-  configurations = ['archives']
-  pkg {
-    repo = "maven"
-    name = "PullRefreshLayout"
-    websiteUrl = siteUrl
-    vcsUrl = gitUrl
-    licenses = ["MIT"]
-    publish = true
-  }
-}
+//apply plugin: 'com.github.dcendents.android-maven'
+//apply plugin: 'com.jfrog.bintray'
+//
+//def repo = new File(rootDir, "repository")
+//
+//apply plugin: 'maven'
+//
+//VERSION_NAME = "1.2.0"
+//def siteUrl = 'https://github.com/baoyongzhang/android-PullRefreshLayout'
+//def gitUrl = 'https://github.com/baoyongzhang/android-PullRefreshLayout.git'
+//GROUP = "com.baoyz.pullrefreshlayout"
+//
+//uploadArchives {
+//  repositories {
+//    mavenDeployer {
+//      repository url: "file://${repo.absolutePath}"
+//      pom.version = VERSION_NAME // version
+//      pom.groupId = GROUP // グループ名
+//      pom.artifactId = POM_ARTIFACT_ID // ライブラリ名
+//    }
+//  }
+//}
+//
+//install {
+//  repositories.mavenInstaller {
+//    // This generates POM.xml with proper parameters
+//    pom {
+//      project {
+//        packaging 'aar'
+//        // Add your description here
+//        name 'This component like SwipeRefreshLayout'
+//        url siteUrl
+//        // Set your license
+//        licenses {
+//          license {
+//            name 'The MIT License (MIT)'
+//            url 'http://baoyz.com/licenses/LICENSE.txt'
+//          }
+//        }
+//        developers {
+//          developer {
+//            id 'baoyongzhang'
+//            name 'baoyongzhang'
+//            email 'baoyz94@gmail.com'
+//          }
+//        }
+//        scm {
+//          connection gitUrl
+//          developerConnection gitUrl
+//          url siteUrl
+//        }
+//      }
+//    }
+//  }
+//}
+//task sourcesJar(type: Jar) {
+//  from android.sourceSets.main.java.srcDirs
+//  classifier = 'sources'
+//}
+//task javadoc(type: Javadoc) {
+//  source = android.sourceSets.main.java.srcDirs
+//  classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
+//}
+//task javadocJar(type: Jar, dependsOn: javadoc) {
+//  classifier = 'javadoc'
+//  from javadoc.destinationDir
+//}
+//artifacts {
+//  archives javadocJar
+//  archives sourcesJar
+//}
+//Properties properties = new Properties()
+//properties.load(project.rootProject.file('local.properties').newDataInputStream())
+//bintray {
+//  user = properties.getProperty("bintray.user")
+//  key = properties.getProperty("bintray.apikey")
+//  configurations = ['archives']
+//  pkg {
+//    repo = "maven"
+//    name = "PullRefreshLayout"
+//    websiteUrl = siteUrl
+//    vcsUrl = gitUrl
+//    licenses = ["MIT"]
+//    publish = true
+//  }
+//}
diff --git a/library/src/main/java/com/baoyz/widget/PullRefreshLayout.java b/library/src/main/java/com/baoyz/widget/PullRefreshLayout.java
index e40ee31..7752161 100644
--- a/library/src/main/java/com/baoyz/widget/PullRefreshLayout.java
+++ b/library/src/main/java/com/baoyz/widget/PullRefreshLayout.java
@@ -25,6 +25,7 @@ import java.security.InvalidParameterException;
  * Created by baoyz on 14/10/30.
  */
 public class PullRefreshLayout extends ViewGroup {
+    private static String TAG = PullRefreshLayout.class.getSimpleName();
 
     private static final float DECELERATE_INTERPOLATION_FACTOR = 2f;
     private static final int DRAG_MAX_DISTANCE = 64;
@@ -160,6 +161,7 @@ public class PullRefreshLayout extends ViewGroup {
         mTarget.measure(widthMeasureSpec, heightMeasureSpec);
         mRefreshView.measure(widthMeasureSpec, heightMeasureSpec);
 //        mRefreshView.measure(MeasureSpec.makeMeasureSpec(mRefreshViewWidth, MeasureSpec.EXACTLY), MeasureSpec.makeMeasureSpec(mRefreshViewHeight, MeasureSpec.EXACTLY));
+        Log.e(TAG, "onMeasure:");
     }
 
     private void ensureTarget() {
@@ -517,6 +519,7 @@ public class PullRefreshLayout extends ViewGroup {
 
         mTarget.layout(left, top + mTarget.getTop(), left + width - right, top + height - bottom + mTarget.getTop());
         mRefreshView.layout(left, top, left + width - right, top + height - bottom);
+        Log.e(TAG, "onLayout:");
     }
 
     private int dp2px(int dp) {
diff --git a/sample/build.gradle b/sample/build.gradle
index d948be4..b0807cd 100644
--- a/sample/build.gradle
+++ b/sample/build.gradle
@@ -7,7 +7,7 @@ android {
 
     defaultConfig {
         applicationId "com.baoyz.pullrefreshlayout.sample"
-        minSdkVersion 8
+        minSdkVersion 9
         targetSdkVersion 21
         versionCode 1
         versionName "1.0"
@@ -22,8 +22,8 @@ android {
 
 dependencies {
     compile fileTree(include: ['*.jar'], dir: 'libs')
-    compile 'com.android.support:support-v4:21.0.0'
-    compile 'com.android.support:recyclerview-v7:21.0.0'
+    compile 'com.android.support:support-v4:25.3.1'
+    compile 'com.android.support:recyclerview-v7:25.3.1'
     compile project(':library')
   compile "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
 }
diff --git a/sample/src/main/java/com/baoyz/pullrefreshlayout/sample/NestRecyclerViewActivity.kt b/sample/src/main/java/com/baoyz/pullrefreshlayout/sample/NestRecyclerViewActivity.kt
index 18870ca..0d699cf 100644
--- a/sample/src/main/java/com/baoyz/pullrefreshlayout/sample/NestRecyclerViewActivity.kt
+++ b/sample/src/main/java/com/baoyz/pullrefreshlayout/sample/NestRecyclerViewActivity.kt
@@ -29,8 +29,11 @@ class NestRecyclerViewActivity : Activity() {
     recyclerView.adapter = ArrayAdapter(this, array)
 
     layout = findViewById(R.id.swipeRefreshLayout) as PullRefreshLayout
+    // NestedScrollViewを挟まない場合はこちらが使えます
+//    recyclerView.viewTreeObserver.addOnScrollChangedListener {
+//      layout.isEnabled = recyclerView.getChildAt(0).top == 0
+//    }
 
-    layout.setTargetView(recyclerView)
     layout.setOnRefreshListener {
       layout.postDelayed(
           { layout.setRefreshing(false) }, 4000)
diff --git a/sample/src/main/res/layout/activity_nest_recycler_view.xml b/sample/src/main/res/layout/activity_nest_recycler_view.xml
index f193ff8..d207472 100644
--- a/sample/src/main/res/layout/activity_nest_recycler_view.xml
+++ b/sample/src/main/res/layout/activity_nest_recycler_view.xml
@@ -1,5 +1,4 @@
 <RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
@@ -11,25 +10,32 @@
       android:layout_width="match_parent"
       android:layout_height="match_parent"
       >
-
-    <RelativeLayout
-        android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:background="@color/blue"
-        >
-    <RelativeLayout
+    <android.support.v4.widget.NestedScrollView
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        android:background="@color/blue"
         >
-      <android.support.v7.widget.RecyclerView
-          android:id="@+id/nestedrecyclerView"
+      <RelativeLayout
           android:layout_width="match_parent"
           android:layout_height="match_parent"
-          android:background="#00ff00"
-          />
-    </RelativeLayout>
-    </RelativeLayout>
+          android:background="@color/blue"
+          android:clickable="true"
+          >
+        <RelativeLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:background="@color/blue"
+            android:clickable="true"
+            >
+          <android.support.v7.widget.RecyclerView
+              android:id="@+id/nestedrecyclerView"
+              android:layout_width="match_parent"
+              android:layout_height="match_parent"
+              android:background="#00ff00"
+              android:nestedScrollingEnabled="false"
+              />
+        </RelativeLayout>
+      </RelativeLayout>
+    </android.support.v4.widget.NestedScrollView>
   </com.baoyz.widget.PullRefreshLayout>
 
 </RelativeLayout>
